pennylane_lightning/core/src/simulators/lightning_qubit/StateVectorLQubit.hpp

```diff
@@ -672,6 +706,164 @@ class StateVectorLQubit : public StateVectorBase<PrecisionT, Derived> {
+
+        dispatcher.applySparseMatrix(kernel, arr, BaseType::getNumQubits(),
+                                     row_map, col_idx, values, wires, inverse);
+    }
+
+    /**
+     * @brief Apply a given sparse matrix directly to the statevector using a
+     * given kernel.
+     *
+     * @param kernel Kernel to run the operation
+     * @param matrix Matrix data (in row-major format).
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
+    template <typename Alloc, typename IndexT = std::size_t>
+    inline void applySparseMatrix(Pennylane::Gates::KernelType kernel,
+                                  const std::vector<IndexT> &row_map,
+                                  const std::vector<IndexT> &col_idx,
+                                  const std::vector<ComplexT, Alloc> &values,
+                                  const std::vector<std::size_t> &wires,
+                                  bool inverse = false) {
+        PL_ABORT_IF(row_map.size() - 1 != exp2(wires.size()),
+                    "The size of matrix does not match with the given "
+                    "number of wires");
+
+        applySparseMatrix(kernel, row_map.data(), col_idx.data(), values.data(),
+                          wires, inverse);
+    }
+
+    /**
+     * @brief Apply a given sparse matrix directly to the statevector using raw
+     * pointers.
+     * @param row_map Pointer to the row map data.
+     * @param col_idx Pointer to the column index data.
+     * @param values Pointer to the values data.
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
+    template <typename IndexT = std::size_t>
+    inline void applySparseMatrix(const IndexT *row_map, const IndexT *col_idx,
+                                  const ComplexT *values,
+                                  const std::vector<std::size_t> &wires,
+                                  bool inverse = false) {
+        using Pennylane::Gates::SparseMatrixOperation;
+
+        PL_ABORT_IF(wires.empty(), "Number of wires must be larger than 0");
+
+        const auto kernel = [n_wires = wires.size(), this]() {
+            return getKernelForSparseMatrix(
+                SparseMatrixOperation::SparseMultiQubitOp);
+        }();
+        applySparseMatrix(kernel, row_map, col_idx, values, wires, inverse);
+    }
+
+    /**
+     * @brief Apply a given sparse matrix directly to the statevector.
+     * @param row_map Vector containing the row map data.
+     * @param col_idx Vector containing the column index data.
+     * @param values Vector containing the values data.
+     * @param wires Wires to apply gate to.
+     * @param inverse Indicate whether inverse should be taken.
+     */
+    template <typename Alloc, typename IndexT = std::size_t>
+    inline void applySparseMatrix(const std::vector<IndexT> &row_map,
```

Comment:
Thanks @AmintorDusko! Do we need all these `applySparseMatrix` methods to support SparseQubitUnitary? I know that we implemented all of them for dense `QubitUnitary` support. I just wanted to know if there's any way to reduce the number of overloaded methods here?